### PR TITLE
Deduplicate model properties and Field/Join from entity

### PR DESCRIPTION
**Pull request by mvorisek**
## BC break

`Field` is no longer bound to an entity, thus methods like `Field::get`, `Field::set` requires entity provided as the 1st argument like `$field->get($entity)`

BC/magic mapping for model properties and methods on entity is provided

### CODE REVIEW

File: src/Field.php [hunk 1]
```diff
@@ -41,6 +43,16 @@ public function __construct(array $defaults = [])
         }
     }
 
+    /**
+     * @param Model $owner
+     */
+    public function setOwner(object $owner)
```
Comment (mvorisek) [inline src/Field.php:49]: DEBUG only, Field should never be bound with an entity

File: src/Model/EntityField.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Data\Model;
+
+use Atk4\Data\Field;
+use Atk4\Data\Model;
+
+class EntityField
```
Comment (mvorisek) [inline src/Model/EntityField.php:10]: test should be added

**Comment by DarkSide666** [comment]
Not so sure about this. Looks like this makes working with models/entities even more complex than it was before.
I'll have to focus and think more about this before approving.

**Comment by mvorisek** [comment]
> 
> 
> Not so sure about this. Looks like this makes working with models/entities even more complex than it was before. I'll have to focus and think more about this before approving.

internally a lot of `getModel(true)` calls must be added to places where model/entity usage is mixed (persistence/joins), from user perspective, there should be no ambiguity - entity should not provide functionality to operate on model

currently, any modifications to model properties on cloned to entity are wrong as this creates discrepancy between model and entity which is undesired as the should be single model/source of truth for any size of it's entities

File: src/Field.php [hunk 3]
```diff
@@ -272,15 +288,10 @@ public function normalize($value)
     /**
      * Casts field value to string.
      *
-     * @param mixed $value Optional value
+     * @param mixed $value
      */
-    public function toString($value = null): string
+    public function toString($value): string
     {
-        $value = ($value === null /* why not func_num_args() === 1 */ ? $this->get() : $this->normalize($value));
```
Comment (mvorisek) [inline src/Field.php:279]: FYI called in typecast too, no need to call twice...

**Review by DarkSide666 — APPROVED** [review]
OK let's go for it in 3.1

**Pull request merged**